app/src/main/java/com/quran/labs/androidquran/util/AudioUtils.kt

```diff
@@ -107,17 +107,25 @@ constructor(private val quranInfo: QuranInfo, private val quranFileUtils: QuranF
     return quranFileUtils.gaplessDatabaseRootUrl + "/" + dbName
   }
 
-  fun getLastAyahToPlay(startAyah: SuraAyah,
-                        currentPage: Int,
-                        mode: Int,
-                        isDualPages: Boolean): SuraAyah? {
-    val page = if (isDualPages && mode == LookAheadAmount.PAGE && currentPage % 2 == 1) {
-      // if we download page by page and we are currently in tablet mode
-      // and playing from the right page, get the left page as well.
-      currentPage + 1
-    } else {
-      currentPage
-    }
+  fun getLastAyahToPlay(
+    startAyah: SuraAyah,
+    currentPage: Int,
+    mode: Int,
+    isDualPages: Boolean,
+    isNotSplitOrShowingTranslation: Boolean
+  ): SuraAyah? {
+    val page =
+      if (isDualPages &&
+        isNotSplitOrShowingTranslation &&
```

Comment:
this isn't right for cases where we are tablet in translation mode without the split setting - in that case, we should still treat it as 2 pages whereas this code will treat it as 1.

Reply:
Will have a look this weekend Insha Allah. Same for all the other comments.

Reply:
Salaam walaikum, I finally got around to this 😅. 

I changed this function along with others with a similar signature to depend on a single boolean that determines if two/one page(s) are/is displayed - 702469884a9d289c237c974edda1aa66bceaf896.